.github/workflows/release.yml

```diff
@@ -116,8 +120,76 @@ jobs:
           --root-dir=${{ github.workspace }}/botan --build-dir=${{ github.workspace }}/build
           --cc='${{ matrix.platform.compiler }}' --make-tool='${{ matrix.platform.make_tool }}'
           --disabled-tests=tls_extensions_key_share_client_hello
+          --test-results-dir=junit_reports
           ${{ matrix.target }}
 
+      - name: Store JUnit Report
+        uses: actions/upload-artifact@v3
+        if: always()
+        with:
+          name: junit
+          path: junit_reports/*.xml
+          if-no-files-found: error
+
+  test_report:
+    name: "Generate Test Report"
+    needs: bsi_tests
+    if: always()
+    runs-on: ubuntu-latest
+    steps:
+      - name: Fetch Audit Repository
+        uses: actions/checkout@v3
+        with:
+          path: ./source
+
+      - name: Setup Environment Configuration
+        uses: ./source/.github/actions/setup-environment
+        with:
+          env_file: ./source/config/botan.env
+
+      - name: Fetch Botan Repository
+        uses: actions/checkout@v3
+        with:
+          path: ./botan
+          repository: ${{ env.BOTAN_REPO }}
+          fetch-depth: 0
+          ref: ${{ env.BOTAN_REF }}
+
+      - name: Botan Git SHA
+        id: vars
+        run: echo "botan_sha=$(git -C botan rev-parse HEAD)" >> $GITHUB_OUTPUT
+
+      - name: Apply Patch to Botan
+        run: git -C botan apply ${{ github.workspace }}/source/docs/testreport/test_properties.patch
```

Comment:
Maybe add a TODO so that we don't forget to remove this. Albeit, it'll probably also fail to apply on newer source versions.

Reply:
That said: It seems to me that we're using the botan checkout exclusively for the Git SHA in this action. Maybe leave this patch out entirely here?